File: Math/2894_Divisible_and_Non-divisible_Sums_Difference/2894.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function(n, m) {
    let num1 = 0;
    let num2 = 0;
    for (let i=1; i<n+1; i++){
       // not divisible
       if (i % m != 0){
           num1 += i; 
       }else{ // divisible
           num2 += i;
       }
    }
    return num1 - num2
};

console.log(differenceOfSums(10,3))

File: Math/2894_Divisible_and_Non-divisible_Sums_Difference/2894.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        num1 = 0
        num2 = 0
        for i in range(1, n+1):
            # not divisible
            if i % m != 0:
                num1 += i
            # divisible
            else:
                num2 += i
        return num1 - num2
    
    def test(self):
        """test function"""
        print(self.differenceOfSums(10, 3))
        print(self.differenceOfSums(5, 6))
        print(self.differenceOfSums(5, 1))

epsilon = Solution()
print(epsilon.test())